spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -303,8 +308,14 @@ object TypeConverter {
   }
 
   implicit object SqlDateConverter extends NullableTypeConverter[java.sql.Date] {
+
+    def subtractTimeZoneOffset(millis: Long) = millis - defaultTimezone.getOffset(millis)
+
     def targetTypeTag = SqlDateTypeTag
-    def convertPF = DateConverter.convertPF.andThen(d => new java.sql.Date(d.getTime))
+
+    def convertPF = PartialFunction[Any, java.sql.Date]{ case x: LocalDate =>
```

Comment:
Either I get it wrong or it will fail unless the object is of `LocalDate` type - is it intended here? Previously `DateConverter.convertPF` was used which is able to handle many different types...